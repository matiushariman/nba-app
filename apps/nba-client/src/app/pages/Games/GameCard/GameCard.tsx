import Scoreboard from './Scoreboard';
import type { GameCardProps } from './GameCard.types';

const GameCard = ({ homeTeam, awayTeam, gameStatusText }: GameCardProps) => (
  <div className="shadow-md cursor-pointer items-center bg-white rounded">
    <Scoreboard
      homeTeam={homeTeam}
      awayTeam={awayTeam}
      gameStatusText={gameStatusText}
    />
  </div>
);

export default GameCard;
