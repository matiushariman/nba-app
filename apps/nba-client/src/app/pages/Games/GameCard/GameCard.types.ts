import type { Team, Game } from '@nba-app/types-nba';

type GameTeam = Pick<Team, 'teamId' | 'teamName' | 'wins' | 'losses' | 'score'>;

export interface GameCardProps
  extends Pick<
    Game,
    'gameStatusText' | 'gameLeaders' | 'gameId' | 'gameStatus'
  > {
  readonly homeTeam: GameTeam;
  readonly awayTeam: GameTeam;
}
