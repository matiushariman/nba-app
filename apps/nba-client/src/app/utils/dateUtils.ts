import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
