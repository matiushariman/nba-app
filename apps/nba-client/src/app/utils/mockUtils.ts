export const shouldUseMock = process.env['NODE_ENV'] === 'development';
