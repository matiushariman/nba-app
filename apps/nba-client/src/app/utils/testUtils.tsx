import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
