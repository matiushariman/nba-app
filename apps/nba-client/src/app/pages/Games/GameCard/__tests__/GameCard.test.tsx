import GameCard from '../GameCard';
import { render } from '../../../../utils/testUtils';

describe('pages/Games/GameCard', () => {
  const homeTeam = {
    teamId: 0,
    teamName: 'Celtics',
    wins: 82,
    losses: 0,
    score: 131,
  };
  const awayTeam = {
    teamId: 1,
    teamName: 'Lakers',
    wins: 0,
    losses: 82,
    score: 92,
  };

  it('should render GameCard that matches snapshot', () => {
    const { baseElement } = render(
      <GameCard
        homeTeam={homeTeam}
        awayTeam={awayTeam}
        gameStatusText="FINAL"
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
