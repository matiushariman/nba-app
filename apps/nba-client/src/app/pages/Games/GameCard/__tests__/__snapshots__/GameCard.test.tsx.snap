// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Games/GameCard should render GameCard that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="shadow-md cursor-pointer items-center bg-white rounded"
    >
      <div
        class="pt-2 pb-2 pl-4 pr-4 flex gap-2"
      >
        <div
          class="flex gap-3 items-center"
        >
          <div
            class="grid gap-1 justify-center"
          >
            <img
              alt="Lakers logo"
              height="72"
              loading="lazy"
              src="https://cdn.nba.com/logos/nba/1/global/L/logo.svg"
              width="72"
            />
            <div>
              <p
                class="text-center"
              >
                Lakers
              </p>
              <p
                class="text-center text-gray-500 text-sm"
              >
                0 - 82
              </p>
            </div>
          </div>
          <div
            class="w-[52px]"
          >
            <p
              class="font-bold text-3xl"
            >
              92
            </p>
          </div>
        </div>
        <div
          class="flex-grow flex justify-center"
        >
          <p
            class="text-sm uppercase font-bold"
          >
            FINAL
          </p>
        </div>
        <div
          class="flex gap-3 items-center"
        >
          <div
            class="w-[52px]"
          >
            <p
              class="font-bold text-3xl"
            >
              131
            </p>
          </div>
          <div
            class="grid gap-1 justify-center"
          >
            <img
              alt="Celtics logo"
              height="72"
              loading="lazy"
              src="https://cdn.nba.com/logos/nba/0/global/L/logo.svg"
              width="72"
            />
            <div>
              <p
                class="text-center"
              >
                Celtics
              </p>
              <p
                class="text-center text-gray-500 text-sm"
              >
                82 - 0
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
